core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -461,7 +475,89 @@ public GeoDistanceSortBuilder fromXContent(QueryParseContext context, String ele
         result.coerce(coerce);
         result.ignoreMalformed(ignoreMalformed);
         return result;
+    }
+
+    @Override
+    public SortField build(QueryShardContext context) throws IOException {
+        final boolean indexCreatedBeforeV2_0 = context.indexVersionCreated().before(Version.V_2_0_0);
+        // validation was not available prior to 2.x, so to support bwc percolation queries we only ignore_malformed on 2.x created indexes
+        List<GeoPoint> localPoints = new ArrayList<GeoPoint>();
+        for (GeoPoint geoPoint : this.points) {
+            localPoints.add(new GeoPoint(geoPoint));
+        }
+
+        if (!indexCreatedBeforeV2_0 && !ignoreMalformed) {
+            for (GeoPoint point : localPoints) {
+                if (point.lat() > 90.0 || point.lat() < -90.0) {
```

Comment:
Why not use this: [GeoUtils.isValidLatitude()](https://github.com/elastic/elasticsearch/blob/52ee4c70275a6b54ac193f5a13e185355dc962bc/core/src/main/java/org/elasticsearch/common/geo/GeoUtils.java#L79)